server/elasticsearch/src/test/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearchTest.java

```diff
@@ -173,16 +186,210 @@ public void testGetSpanFromEmptyDB() throws StoreException, InterruptedException
         Assert.assertNull(spanFromDB);
     }
 
-    private void storeAndWait(String tenant, Span span) throws StoreException, InterruptedException {
-        spanService.storeSpan(tenant, Arrays.asList(span));
+    @Test
+    public void testGetTraceFragmentRootServerSpan() throws StoreException, InterruptedException {
+        Span rootServerSpan = new Span();
+        rootServerSpan.setId("root");
+        rootServerSpan.setTraceId("root");
+        rootServerSpan.setAnnotations(serverAnnotations());
+
+        Span clientDescendant = new Span();
+        clientDescendant.setId("descendant");
+        clientDescendant.setParentId("root");
+        clientDescendant.setTraceId("root");
+        clientDescendant.setAnnotations(clientAnnotations());
+
+        Span clientDescendant2 = new Span();
+        clientDescendant2.setId("descendant2");
+        clientDescendant2.setParentId("root");
+        clientDescendant2.setTraceId("root");
+        clientDescendant2.setAnnotations(clientAnnotations());
+
+        Span serverDescendantOfDescendant2 = new Span();
+        serverDescendantOfDescendant2.setId("descendant2");
+        serverDescendantOfDescendant2.setParentId("descendant2");
```

Comment:
This should be setParentId("root") - should be same value as the associated clientDescendant2